lib/Net/Google/Drive/Simple/V3.pm

```diff
@@ -0,0 +1,2475 @@
+    my ( $self, $options, $body_param_names ) = @_;
+
+    my $body_options = {
+        map +( exists $options->{$_} ? ( $_ => delete $options->{$_} ) : () ),
+        @{$body_param_names},
+    };
+
+    return $body_options;
+}
+
+###########################################
+sub _handle_api_method {
+###########################################
+    my ( $self, $info, $options ) = @_;
+    my $method = $info->{'method_name'};
+
+    # We yank out all the body parameters so we don't validate them
+    # TODO: Support body parameter validation
+    my $body_options = $self->_prepare_body_options( $options, $info->{'body_parameters'} );
+
+    # We validate the options left
+    $self->_validate_param_type( $method, $info->{'query_parameters'}, $options );
+
+    # We might have a more complicated type checking specified
+    if ( my $param_check = $info->{'parameter_checks'} ) {
+        foreach my $name ( sort keys %{$param_check} ) {
+            defined $options->{$name}
+                or next;
+
+            my $cb = $param_check->{$name};
+            ref $cb eq 'CODE'
+                or LOGDIE("[$method] Parameter '$name' missing validation callback");
+
+            local $_ = $options->{$name};
+            my $error_str = $cb->();
+            $error_str
+                and LOGDIE("[$method] Parameter '$name' failed validation: $error_str");
+        }
+    }
+
+    # We check for deprecated parameters (from a list of known deprecated parameters)
+    $self->_handle_deprecated_params( $method, $info, $options );
+
+    # We handle some more specific validation rules (from a list of known parameters)
+    $self->_handle_complex_types( $method, $options );
+
+    $self->init();
+
+    # We generate the URI path
+    my $uri = $self->_generate_uri( $info->{'path'}, $options );
+
+    # GET requests cannot have a body
+    if (   $info->{'http_method'} eq HTTP_METHOD_GET()
+        || $info->{'http_method'} eq HTTP_METHOD_DELETE() )
+    {
+        undef $body_options;
+    }
+
+    # We make the request and get a response
+    return $self->http_json( $uri, [ $info->{'http_method'}, $body_options ] );
+}
+
+# --- about
+
+###########################################
+sub about {
+###########################################
+    my ( $self, $options ) = @_;
+
+    ref $options eq 'HASH'
+        or LOGDIE('about() missing parameters');
+
+    my $info = {
+        'query_parameters' => {
+            'fields' => [ TYPE_STRING(), 1 ],
+        },
+        'path'        => 'about',
+        'method_name' => 'about',
+        'http_method' => HTTP_METHOD_GET(),
+    };
+
+    return $self->_handle_api_method( $info, $options );
+}
+
+# --- changes
+
+###########################################
+sub getStartPageToken {
+###########################################
+    my ( $self, $options ) = @_;
+
+    $options //= {};
+
+    my $info = {
+        'query_parameters' => {
+            'driveId'            => [ TYPE_STRING(),  0 ],
```

Comment:
note: we are discarding constant folding by calling the constant as a CV?

Reply:
No, constant folding will still occur. This is just a clearer indication to the reader (and the parser) that it's a CV. The optimizer identifies the CV is a constant and will fold it just the same.

```
$ perl -MO=Concise -E'sub FOO () {1} if ( FOO ) { say "ok" }'
6  <@> leave[1 ref] vKP/REFC ->(end)
1     <0> enter v ->2
2     <;> nextstate(main 4 -e:1) v:%,us,{,fea=7 ->3
-     <@> scope vK/FOLD ->6
-        <;> ex-nextstate(main 6 -e:1) v:%,us,fea=7 ->3
5        <@> say vK ->6
3           <0> pushmark s ->4
4           <$> const[PV "ok"] s ->5
-e syntax OK

$ perl -MO=Concise -E'sub FOO () {1} if ( FOO() ) { say "ok" }'
6  <@> leave[1 ref] vKP/REFC ->(end)
1     <0> enter v ->2
2     <;> nextstate(main 4 -e:1) v:%,us,{,fea=7 ->3
-     <@> scope vK/FOLD ->6
-        <;> ex-nextstate(main 6 -e:1) v:%,us,fea=7 ->3
5        <@> say vK ->6
3           <0> pushmark s ->4
4           <$> const[PV "ok"] s ->5
-e syntax OK
```